src/api/index.js

```diff
@@ -300,6 +300,40 @@ export async function fetchMarkRepoNotificationAsRead(
   return response;
 }
 
+export async function fetchNotificationsCount(accessToken) {
+  const ENDPOINT = `${root}/notifications?per_page=1`;
+  const response = await fetch(ENDPOINT, accessTokenParameters(accessToken));
+
+  let linkHeader = response.headers.get('Link');
+  let count = 0;
+
+  if (linkHeader) {
+    linkHeader = linkHeader.match(/page=(\d)+/g).pop();
+    count = linkHeader.split('=').pop();
+  }
+
+  return count;
+}
+
+export async function fetchRepoNotificationsCount(
+  owner,
+  repoName,
+  accessToken
+) {
+  const ENDPOINT = `${root}/repos/${owner}/${repoName}/notifications?per_page=1`;
+  const response = await fetch(ENDPOINT, accessTokenParameters(accessToken));
+
+  let linkHeader = response.headers.get('Link');
+  let count = 1;
```

Comment:
if the Link header is not present, count should be the count of json objects returned in the response, not 1

Reply:
See how this is handled in v3.count() I just added: https://github.com/gitpoint/git-point/pull/361/commits/6ce848985be790541f1cb2029e7a061dca47226b#diff-284817f1b6aad47c0ac6d61bd72f474cR58

Reply:
> if the Link header is not present, count should be the count of json objects returned in the response, not 1
  
The fact is that when the notification in the repository is only one, the Link header will not be, and `response.json().length` will return 0, which is not correct. Therefore, the default is 1.

Reply:
> See how this is handled in v3.count() I just added: 6ce8489#diff-284817f1b6aad47c0ac6d61bd72f474cR58
  
Can you add a third parameter, such as `defaultCount`?

Reply:
@lex111 as I understand, here's how the API respond : 
- if you have 0 notifications, response.status is 404 with a json body that must be ignored
- if you have notifications : 
  - if only 1, no Link header and response.json() is an array of 1 element (so json().length = 1)
  - if more, you parse the Link header to get the number

am I missing something?

Reply:
Yes, that's right, I checked again,  and found the error that [you have now](https://github.com/gitpoint/git-point/pull/361/commits/6ce848985be790541f1cb2029e7a061dca47226b#diff-284817f1b6aad47c0ac6d61bd72f474cR67), and because of which I incorrectly put 1

`response.json()` is a promise, so it should be like this:

```js
number = response.json().then(data => {
  return data.length;
});
```

Reply:
OMG, thanks! didn't know response.json() was a promise, I thought only response was.
I'll update count() to handle the 404, fix json().
Do you still think defaultValue is needed? I fail to see how

Reply:
> Do you still think defaultValue is needed? I fail to see how

No, at least for now.

Reply:
just fixed the `count()` method and realized that json().length isn't needed at all as we request one element per page -_-

So, if 404 => 0 elements
If no Link header => 1 element
Else, count fetched from Link

I wonder if the 404 status code is sent for all empty request responses ..

Reply:
(let's chat over gitter if you want)